### PR TITLE
Fix #51067 create_table if_not_exists guard S prevent force drop_table

### LINKED ISSUE #51067: create_table `force` option drops table when `if_not_exists` option is present
The problem is here:
https://github.com/rails/rails/blob/36c1591bcb5e0ee3084759c7f42a706fe5bb7ca7/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L298-300

I can't submit this from my work computer, but can somebody put up a PR to make this change:
```ruby
drop_table(table_name, force: force, if_exists: true) unless options[:if_not_exists]
```

Edge case where this occurs:
Adding guarded migrations to make migration schema consistent with schema dump for DB that has migrations applied to it that are not in the schema history.
If you copy the table definition from the schema dump and add the `if_not_exists`, running the migration will drop the table if it exists

### PR DESCRIPTION
**Pull request by allisonphillips**
Solving https://github.com/rails/rails/issues/51067

I'm only familiar with writing rails tests in rspec, I wrote a test after skimming the other syntax in the file but please let me know if there is a better way. `record.reload` should raise an exception if the record was deleted by `create_table` dropping the table.

> [!NOTE]
> Currently limited to a web editor so I don't have the ability to squash commits or run the full test suite, will try to grab some time to clean up the branch and validate tests when my personal machine is accessible

### Motivation / Background

This Pull Request has been created because https://github.com/rails/rails/issues/51067

### Detail

This Pull Request changes `create_table` to ignore the `force` option (which drops the table) if the `if_not_exists` option is present

### Additional information

If you copy a `create_table` migration from a schema dump and add the `if_not_exists` guard but forget to remove `force`, it will drop your table when you migrate it. 

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb [hunk 1]
```diff
@@ -296,7 +296,7 @@ def create_table(table_name, id: :primary_key, primary_key: nil, force: nil, **o
         td = build_create_table_definition(table_name, id: id, primary_key: primary_key, force: force, **options, &block)
 
         if force
```
Comment (adrianna-chang-shopify) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb:298]: Should we move the check for `if_not_exists` up?
```suggestion
        if force && !td.if_not_exists
```

Previously, if `:force` and `:if_not_exists` were both set, we would've dropped the table and cleared the schema cache. If only `:if_not_exists` was set, we would've fallen into the `else` branch and cleared the schema cache. Moving the check up ensures that we still fall into the `else` branch and clear the schema cache.

File: activerecord/test/cases/migration_test.rb [hunk 2]
```diff
@@ -193,6 +193,15 @@ def test_create_table_with_if_not_exists_true
         t.string :foo
       end
     end
+
+    # Do not drop table if existence guard option is present
+    assert_nothing_raised do
```
Comment (adrianna-chang-shopify) [inline activerecord/test/cases/migration_test.rb:198]: Can we extract this to its own test? Maybe `test_create_table_with_if_not_exists_true_and_force_true`

**Review by adrianna-chang-shopify — COMMENTED** [review]
I'm wondering if it would be worth raising an `ArgumentError` instead. While I agree that giving priority to `if_not_exists` is preferable to dropping the table,`force` and `if_not_exists` are conflicting options, so it might be a better user experience to raise if both are supplied.

**Comment by skipkayhil** [comment]
Merged in #51513, thanks!

**Pull request closed without merging**